### PR TITLE
Fix Windows prompt refinement: ensure 'bash' is replaced with 'powershell' in all prompts

### PR DESCRIPTION
**Pull request by li-boxuan**
- [ ] This change is worth documenting at https://docs.all-hands.dev/
- [x] Include this change in the Release Notes. If checked, you **must** provide an **end-user friendly** description for your change below

**End-user friendly description of the problem this fixes or functionality this introduces.**

Fixed an issue where CodeActAgent system prompts and in-context learning examples contained references to "bash" on Windows platforms, which should be replaced with "powershell" for consistency with the Windows runtime environment. This ensures that Windows users see appropriate shell references in all agent communications.

---
**Summarize what the PR does, explaining any non-trivial design decisions.**

This PR addresses missing applications of the `refine_prompt` function in the Windows runtime. The `refine_prompt` function was designed to replace "bash" references with "powershell" on Windows, but it was only being applied to tool descriptions, not to system prompts or in-context learning examples.

**Changes made:**

1. **Enhanced `refine_prompt` function** (`openhands/agenthub/codeact_agent/tools/bash.py`):
   - Improved to handle case-insensitive replacement with proper word boundaries
   - Properly handles "execute_bash" → "execute_powershell" transformation
   - Preserves words like "bashful" that contain "bash" but aren't the word "bash"
   - Fixed misleading comment

2. **Applied `refine_prompt` to system messages** (`openhands/utils/prompt.py`):
   - Modified `PromptManager.get_system_message()` to apply `refine_prompt` to rendered system messages

3. **Applied `refine_prompt` to in-context learning examples** (`openhands/llm/fn_call_converter.py`):
   - Modified `get_example_for_tools()` to apply `refine_prompt` to generated examples

4. **Added comprehensive unit tests** (`tests/unit/test_windows_prompt_refinement.py`):
   - Tests that run only on Windows (skipped on other platforms)
   - Verifies system prompts, tool descriptions, and in-context learning examples don't contain "bash" on Windows
   - Tests the `refine_prompt` function directly with various edge cases

**Design decisions:**
- Used regex with word boundaries and negative lookbehind to ensure precise replacement
- Applied refinement at the final rendering stage to catch all prompt content
- Made tests Windows-specific using `pytest.mark.skipif` to avoid false failures on other platforms

---
**Link of any specific issues this addresses:**

This addresses an internal consistency issue where Windows users would see mixed "bash" and "powershell" references in agent prompts, which could be confusing.

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:ba5c635-nikolaik   --name openhands-app-ba5c635   docker.all-hands.dev/all-hands-ai/openhands:ba5c635
```

CLI with uvx:
```
uvx --python 3.12 --from git+https://github.com/All-Hands-AI/OpenHands@fix/windows-prompt-refinement-bash-replacement openhands
```

### CODE REVIEW

File: openhands/agenthub/codeact_agent/tools/bash.py [hunk 1]
```diff
@@ -37,7 +37,18 @@
 
 def refine_prompt(prompt: str):
     if sys.platform == 'win32':
-        return prompt.replace('bash', 'powershell')
+        # Replace 'bash' with 'powershell' including tool names like 'execute_bash'
+        import re
```
Comment (enyst) [inline openhands/agenthub/codeact_agent/tools/bash.py]: Nit: could this import be on top of the file?

Comment (li-boxuan) [reply openhands/agenthub/codeact_agent/tools/bash.py]: yeah claude really likes to put import inline!

@openhands please move the import to the top

**Review by enyst — APPROVED** [review]
Thank you!
